### ISSUE TITLE
Allow modifying staged files while using git full editor

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#7938]
I like to review my files in detail from SCM panel while creating the git message. Often during this process I find very minor - that I like to change; things like: rename variable, modify comment, minor refactor, etc. It's easy enough to make these changes and run the test suite while still in the git full editor experience. The problem is SCM doesn't refresh nor include changes that are made.

Workflow:
- stage changes
- press SCM check mark to start git full editor
- begin editing message while reviewing diffs
- make small change, save file, run test suite
- notice SCM panel did not refresh to show unstaged changes, press SCM refresh button
- changed file is now in SCM unstaged changes area, click add button
- change seems to be staged
- finish git message, save and close git full editor
- changes that were staged when starting commit are committed, SCM refreshes to show newly staged changes still there and not commited

Specific changes for above workflow:
- SCM panel should refresh as usual during git full editor experience.
- Newly staged changes should be included in git commit when closing git full editor.

As an example, today I made a refactor and during review I noticed a section where I no longer needed a cast. After making the change, manually refreshing SCM, staging the file, finishing my commit message, and finally closing the git full editor, the SCM refreshed to show my change remained behind and was not committed.
![image](https://user-images.githubusercontent.com/797824/189439243-236b50ac-58a0-40f0-bbd9-e37bc79a02c1.png)

I suppose this could be avoided by reviewing the files in detail before starting git commit, but if you create detailed git messages, then you'd have to basically review the files in detail a second time to fill out the git message.
